src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1419,6 +1662,271 @@ DatabaseShardImp::removeFailedShard(std::shared_ptr<Shard> shard)
     setFileStats();
```

Comment:
Also not in this changeset, but what is the point of the `reset` on the previous line? If there is one it should be documented, if there isn't it should be removed.

Reply:
This causes the shard to be removed from the `DatabaseShardImp`. This is why it's important for the shard to be passed by reference.

Reply:
Added a comment.